src/layout/Grid/GridComponent.tsx

```diff
@@ -76,29 +82,48 @@ export function GridRowRenderer({ row, isNested, mutableColumnSettings }: GridRo
           mutableColumnSettings[cellIdx] = cell.columnOptions;
         }
 
-        if (cell && 'text' in cell && cell.text) {
+        const closestComponent = () => {
+          if (cell && 'labelFrom' in cell && cell?.labelFrom) {
+            const idToFind = cell.labelFrom;
+            return node.flat(true).find((n) => n.item.id === idToFind || n.item.baseComponentId === idToFind);
+          }
+        };
```

Comment:
Nitpicking here, but this code could probably just be moved down into the other `if` below, and it has no need to be a function, so you could just pass the found node to `<CellWithLabel>` directly.